Automattic/woocommerce-payments
# Add Customer Currency filter to transactions advanced filters

Fixes #1278

#### Changes proposed in this Pull Request



- Adds a way to fetch a list of all the currencies that have ever been used in a customer's store
- Adds a filter to allow filtering by customer currency.







#### Testing instructions





- Checkout 2176-gh-Automattic/woocommerce-payments-server on your local server and run `npm run listen`.
-`npm run watch` on the client
- Make a couple of transactions on your store in various currencies.
- Go to `Payments > Transactions`, open the advanced filters menu and test the `Customer currency` filter, using both the `is` and `is not` filters, ensuring you get the expected result (only orders made in the selected currency should be shown, and when using `is not` only orders made in a different currency from the one selected should be shown.
- Test the other advanced filters to ensure they still work as expected

Will add tests, but wanted to get feedback on the proposed solution first.

#### Questions

- I added the 'Is Not' filter as well. I was thinking this could be useful to see all transactions not in the default store currency, for example. Do we want this?
- I tried to refactor the logic to make it more efficient, but there's a few things I'm unsure about. It would be nice if the customer currencies list is only populated on first page load (rather than being repopulated every time a search is made) since these values will likely not change after the initial load. (although, in theory, more transactions could be made in this time and new currencies may be added to the list).
- Also, the deposit currency filter option repopulates every time the search is made, which I don't think is necessary (to observe this, change the second argument passed to `getFilters` to `true` in the `transactions/filters/index.tsx` file.

-------------------


**Post merge**
